core/src/main/java/com/sequenceiq/cloudbreak/converter/v4/stacks/instancegroup/InstanceGroupV4RequestToInstanceGroupConverter.java

```diff
@@ -52,6 +58,13 @@ private void addInstanceMetadatas(InstanceGroupV4Request request, InstanceGroup
 
     private void setAttributes(InstanceGroupV4Request source, InstanceGroup instanceGroup) {
         Map<String, Object> parameters = providerParameterCalculator.get(source).asMap();
+        if (source.getCloudPlatform() == CloudPlatform.AZURE) {
```

Comment:
This was intentional. The Basic SKU Azure load balancer can only route traffic to instances it shares an availability set with. However, we were not previous grouping instances in the stack into availability sets. So this logic was added to group instances such that each instance group has its own availability set, so that we can add the load balancer to the appropriate availability.

Reply:
If that is the case, pls consult with @doktoric as we removed AS setting by default

Reply:
@brycederriso has refactored this approach so that instead of always putting instances in an availability set, we only create an availability set that contains the instances the load balancer will route traffic to. Bryce has discussed this new approach with Richard and gotten approval for it. I believe he's testing it now. @brycederriso please provide an update.

Reply:
I'm working on selectively adding the Availability set to an instance group if it actually contains Knox Load Balancing targets.

Reply:
I've moved this logic into the `StackV4RequestToStackConverter`. We only add an availability set based on target groups now.

Please see `StackV4RequestToStackConverter.configureLoadBalancerAvailabilitySets()`.

Reply:
@brycederriso  Just out of curiosity what lead to the decision to go with Basic SKU instead of Standard, besides the price?